debbuild

```diff
@@ -1812,181 +1970,522 @@ sub store_value {
+        $m = expd_simple('patch'.$arg);
+      }
+    } elsif ( my ($macra,$spc,$args) = $m =~ /\A\{(\w+)(\s+)(.+)\}\z/s ) {
+      $m = expandmacros(handle_macro_options("%$macra $args"));
+    } elsif ( my ($smpl) = $m =~ /\A\{\$?([a-z_]\w*)\s*\}\z/i ) {
+      $m = expd_simple($smpl);
+    }
+    vdebug($o, 5, "expandmacros/complex/in");
+    my $i = scalar @macropsstk;
+    vdebug("Parameterized macro stack: [".($i)."]", 7, "expandmacros/complex/stack");
+    foreach my $ms ( @macropsstk ) {
+      vdebug("[$i] ".$ms->{'0'}." / ".$ms->{'#'}.": ".$ms->{'**'}, 7, "expandmacros/complex/stack");
+      $i--;
+    }
+    vdebug($m, 5, "expandmacros/complex/out");
+    return ($m eq $o) ? '%'.$m : $m;
   }
-  pos = 0; # reset to start of $_
-
-  # unescape multi-line macros
-  s/\\"/"/g;
-  s/\\\\(\w)/\\$1/g;
-  s/(\S)\\+\n/$1\n/g;
-  # revert changes of dir macro because of spcial use in %files
-  s/%\{dir\}/%dir/g;
-  return $_;
+  sub expd_exec {
+    my ($c) = @_;
+    my $r = '';
+    $c =~ s/\\\$/\$/g;
+    $c =~ s/\\\\n/\\n/g;
+    open(my $ch, '-|', $build_shell, '-c', $c) or return $r;
+    while ( <$ch> ) {
+      $r .= $_;
+    }
+    close($ch);
+    chomp($r);
+    return $r;
+  }
+  while ( $l ) {
+    if ( $s =~ /%+/ ) {
+      $r .= $`;
+      $s = $';
+      my $qs = $&;
+      my $ql = length $qs;
+      $l -= length($`) + $ql;
+      $r .= '%' x ($ql-1);
+      next if $ql > 1;
+      if ( $s =~ /\A[\w\*\#]+/ ) {
+        my $m = $&;
+        $s = $';
+        $l -= length $m;
+        if ( $m =~ /\A(description|dir|doc|exclude|ghost|package|if(?:n?arch|)|else|endif|post(?:un)?|pre(?:un)?)\z/ ) {
+          $r .= "%".$m;
+        } else {
+          $r .= expd_simple($m);
+        }
+      } elsif ( $s =~ /\A\{/ ) {
+        my $m = '{';
+        $s = $';
+        my $tr;
+        do {
+          if ( $s =~ /\}/ ) {
+            $m .= $`.$&;
+            $s = $';
+            my $tst = $m;
+            $tr = defined(extract_bracketed($tst, '{}'));
+            if ( $tr ) {
+              $r .= expd_complex($m);
+              $l -= length $m;
+            }
+          } else {
+            $r .= '%'.$m;
+            $l -= length $m;
+            $tr = 1;
+          }
+        } until $tr;
+      } elsif ( $s =~ /\A\(/ ) {
+        my $m = '(';
+        $s = $';
+        my $tr;
+        do {
+          if ( $s =~ /\)/ ) {
+            $m .= $`.$&;
+            $s = $';
+            my $tst = $m;
+            $tr = defined(extract_bracketed($tst, '()'));
+            if ( $tr ) {
+              $m =~ s/\A\(//;
+              $m =~ s/\)\z//;
+              $r .= expd_exec(expandmacros($m));
+              $l -= length $m;
+            }
+          } else {
+            $r .= '%'.$m;
+            $l -= length $m;
+            $tr = 1;
+          }
+        } until $tr;
+      } else {
+        $r .= '%';
+      }
+    } else {
+      $r .= $s;
+      $l = 0;
+      $s = '';
+    }
+  }
+  $r =~ s/\\\\(\w)/\\$1/g;
```

Comment:
Hi @Conan-Kudo . I've updated the PR with adding this only line. It was taken from original `expandmacros`. From my point of view it's not really safe to do it everywhere, but I've tested with all the packages I'm using `debbuild` for and everything looks fine.